frontend/src/metabase/visualizations/lib/scalar_utils.ts

```diff
@@ -0,0 +1,32 @@
+import { formatValue } from "metabase/lib/formatting";
+import { OptionsType } from "metabase/lib/formatting/types";
+
+// used below to determine whether we show compact formatting
+export const COMPACT_MAX_WIDTH = 250;
+export const COMPACT_WIDTH_PER_DIGIT = 25;
+export const COMPACT_MIN_LENGTH = 6;
+
+export function compactifyValue(
+  formatOptions: OptionsType = {},
```

Comment:
Parameters with default arguments should be put last otherwise there's no way to omit them while including the other values (e.g. you wouldn't be able to call `compactifyValue` with `value` and `width` but no `formatOptions`).